src/main/java/frc/robot/robot_manager/RobotManager.java

```diff
@@ -1,18 +1,207 @@
+      case INTAKING -> {
+        if (intake.hasNote() || queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_WITH_GP);
+        }
+      }
+      case OUTTAKING -> {
+        if (!intake.hasNote() && !queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+    }
+
+    return currentState;
+  }
+
+  @Override
+  protected void afterTransition(RobotState newState) {
+    switch (newState) {
+      case SPEAKER_PREPARE_TO_SCORE, SPEAKER_SCORING, SPEAKER_WAITING -> {
+        arm.setState(ArmState.SPEAKER_SHOT);
+        shooter.setState(ShooterState.SPEAKER_SHOT);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_WITH_GP);
+      }
+      case AMP_PREPARE_TO_SCORE, AMP_SCORING, AMP_WAITING -> {
+        arm.setState(ArmState.AMP);
+        shooter.setState(ShooterState.AMP);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_WITH_GP);
+      }
+      case FEEDING_PREPARE_TO_SHOOT, FEEDING_SHOOTING, FEEDING_WAITING -> {
+        arm.setState(ArmState.FEEDING);
+        shooter.setState(ShooterState.FEEDING);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_WITH_GP);
+      }
+      case PASS_PREPARE_TO_SHOOT, PASS_SHOOTING, PASS_WAITING -> {
+        arm.setState(ArmState.PASS);
+        shooter.setState(ShooterState.PASS);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_WITH_GP);
+      }
+      case UNJAM -> {
+        arm.setState(ArmState.AMP);
+        shooter.setState(ShooterState.DROP);
+        intake.setState(IntakeState.OUTTAKING);
+        queuer.seState(QueuerState.OUTTAKING);
+      }
+      case INTAKING -> {
+        arm.setState(ArmState.IDLE);
+        shooter.setState(ShooterState.IDLE_STOPPED);
+        intake.setState(IntakeState.INTAKING);
+        queuer.seState(QueuerState.INTAKING);
+      }
+      case OUTTAKING -> {
+        arm.setState(ArmState.IDLE);
+        shooter.setState(ShooterState.IDLE_STOPPED);
+        intake.setState(IntakeState.OUTTAKING);
+        queuer.seState(QueuerState.OUTTAKING);
+      }
+      case CLIMBING_1_LINEUP -> {
+        arm.setState(ArmState.CLIMBING_1_LINEUP);
+        shooter.setState(ShooterState.IDLE_STOPPED);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_NO_GP);
+      }
+      case CLIMBING_2_HANGING -> {
+        arm.setState(ArmState.CLIMBING_1_LINEUP);
+        shooter.setState(ShooterState.IDLE_STOPPED);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_NO_GP);
+      }
+      case IDLE_NO_GP -> {
+        arm.setState(ArmState.IDLE);
+        shooter.setState(ShooterState.IDLE_STOPPED);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_NO_GP);
+      }
+      case IDLE_WITH_GP -> {
+        arm.setState(ArmState.IDLE);
+        shooter.setState(ShooterState.IDLE_STOPPED);
```

Comment:
i think this is one of the cases where we want to use `IDLE_WARMUP` for the shooter, to start spinning the wheels in case we want to shoot soon